### PR TITLE
arcanist: generate & install bash-completion

### PR DESCRIPTION
**Pull request by benley**
Arcanist only supports bash for shell-completion as of 20200711, otherwise I would include zsh and/or fish here.

Upstream discussion of zsh non-support: https://secure.phabricator.com/D19700

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/arcanist/default.nix [hunk 1]
```diff
@@ -49,6 +50,10 @@ stdenv.mkDerivation {
 
     ${makeArcWrapper "arc"}
     ${makeArcWrapper "phage"}
+
+    $out/bin/arc shell-complete --generate --
+    installShellCompletion --bash --name arc $out/libexec/arcanist/support/shell/rules/bash-rules.sh
+    installShellCompletion --bash --name phage $out/libexec/arcanist/support/shell/rules/bash-rules.sh
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/arcanist/default.nix:56]:
```suggestion
    installShellCompletion --cmd arc --bash $out/libexec/arcanist/support/shell/rules/bash-rules.sh
    installShellCompletion --cmd phage --bash $out/libexec/arcanist/support/shell/rules/bash-rules.sh
```

File: pkgs/development/tools/misc/arcanist/default.nix [hunk 2]
```diff
@@ -29,26 +30,37 @@ stdenv.mkDerivation {
...
 
   postPatch = lib.optionalString stdenv.isAarch64 ''
     substituteInPlace support/xhpast/Makefile \
       --replace "-minline-all-stringops" ""
   '';
 
   buildPhase = ''
+    runHook preBuild
     make cleanall -C support/xhpast
     make xhpast -C support/xhpast
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/arcanist/default.nix:46]: Missing $makeFlags/$NIX_BUILD_CORES.

Comment (benley) [reply pkgs/development/tools/misc/arcanist/default.nix:46]: should I add those in this pull request, or would a separate cleanup change be better?

Comment (SuperSandro2000) [reply pkgs/development/tools/misc/arcanist/default.nix:46]: Just do it in this PR. We are cleaning up anyway and this saves a hydra rebuild cycle.

Comment (benley) [reply pkgs/development/tools/misc/arcanist/default.nix:46]: cool, coming right up.

**Comment by benley** [comment]
Any objection to merging this now?

**Pull request merged**